zulip/integrations/slack/slackdata2zulipdata.py

```diff
@@ -210,33 +210,56 @@ def channels2zerver_stream(slack_dir, realm_id, added_users):
 
 def channelmessage2zerver_message(slack_dir, channel, added_users, added_channels):
     json_names = os.listdir(slack_dir + '/' + channel)
+    users = json.load(open(slack_dir + '/users.json'))
     zerver_message = []
     msg_id_count = 1
+
+    #Sanitize the message text
+    def sanitize(string):
+        save_string = string
+        string = string.split(' ')
+        string[0] = string[0].replace('<@', ' ')
+        string[0] = string[0].replace('>', ' ')
+        string[0] = string[0].replace('|', ' ')
+        length = len(string[0].split(' '))
+        if length > 1:
+            short_name = string[0].split(' ')[2]
+            string[0] = string[0].split(' ')[1]
+        for user in users:
+            if(user['id'] == string[0] and user['name'] == short_name and length == 4) or
+              (user['id'] == string[0] and length == 3):
+                string[0] = user.get('real_name', user['name'])
+                string[0] = "@**" + string[0] + "** "
+                string = " ".join(string)
+                return string
+        return save_string
+
     for json_name in json_names:
         msgs = json.load(open(slack_dir + '/%s/%s' % (channel, json_name)))
         for msg in msgs:
             text = msg['text']
-            try:
-                user = msg.get('user', msg['file']['user'])
-            except KeyError:
-                # black magic, explain this later TOODOO
-                user = msg['user']
-            zulip_message = dict(
-                sending_client=1,
-                rendered_content_version=1,  # This is Zulip-specific
-                has_image=msg.get('has_image', False),
-                subject=channel,  # This is Zulip-specific
-                pub_date=msg['ts'],
-                id=msg_id_count,
-                has_attachment=False,  # attachment will be posted in the subsequent message; this is how Slack does it, less like email
-                edit_history=None,
-                sender=added_users[user],  # map slack id to zulip id
-                content=text,  # TODO sanitize slack text, which contains <@msg['user']|short_name>
-                rendered_content=text,  # slack doesn't cache this
-                recipient=added_channels[channel],
-                last_edit_time=None,
-                has_link=msg['has_link'])
-            zerver_message.append(zulip_message)
+            if "has joined the channel" not in text:
```

Comment:
```python
if "has joined the channel" in text:
    continue
```

achieves the same effect, and doesn't require the subsequent code-blocks to be further indented.